lib/palletjack/tool.rb

```diff
@@ -0,0 +1,286 @@
+require 'palletjack'
+require 'fileutils'
+require 'optparse'
+require 'singleton'
+require 'rugged'
+
+class PalletJack
+
+  # Superclass for PalletJack tool implementations
+  #
+  # Provides convenience methods for option parsing, file generation,
+  # and warehouse structure management.
+  #
+  # Example:
+  #   class MyTool < PalletJack::Tool
+  #     def parse_options(parser)
+  #       parser.on('-o DIR', '--output DIR',
+  #                 'output directory',
+  #                 String) { options[:output] = dir }
+  #
+  #       required_option :output
+  #     end
+  #   end
+  #
+  #   MyTool.run do
+  #
+  #     jack['system'].each do |sys|
+  #        config_dir :output, sys.name
+  #        config_file :output, sys.name, "dump.yaml" do |file|
+  #          file << sys.to_yaml
+  #        end
+  #     end
+  #   end
+
+  class Tool
+    include Singleton
+
+    # Run the +block+ given in the context of the tool singleton instance
+    # as convenience for simple tools.
+    #
+    # More complex tools probably want to override initialize to
```

Comment:
`#initialize` is special in rdoc. Also the rdoc is lying, overriding `#initialize` is not the correct way to do option parsing; `#parse_options` is the correct way.